web3/providers/websocket/request_processor.py

```diff
@@ -177,15 +177,15 @@ def get_request_information_for_response(
 
     def append_middleware_response_processor(
         self,
+        response_id: int,
         middleware_response_processor: Callable[..., Any],
     ) -> None:
-        request_id = next(copy(self._provider.request_counter)) - 1
-        cache_key = generate_cache_key(request_id)
-        current_request_cached_info: RequestInformation = (
+        cache_key = generate_cache_key(response_id)
```

Comment:
In the examples we were walking through earlier, I think there was a sticky point that some responses didnt appear to include an `id`. If that happens, `generate_cache_key` will raise. Should we swallow that exception and just skip the processors here instead?

Reply:
That's a good point. We could still return the raw response and try to not halt things here. I think passing in the `response` to this method would allow us to retrieve the `id` if it's there or debug log the response that doesn't have an `id` and keep moving.

Reply:
Yeah that sounds like it would work great!